src/ristretto_point.rs

```diff
@@ -0,0 +1,218 @@
+use crate::util::nonzero_gadget;
+use bulletproofs::r1cs::{
+    ConstraintSystem, LinearCombination, R1CSError, RandomizedConstraintSystem, Variable,
+};
+use curve25519_dalek::{ristretto::CompressedRistretto, scalar::Scalar};
+use zerocaf::ristretto::RistrettoPoint as SonnyRistrettoPoint;
+use zerocaf::scalar::Scalar as SonnyScalar;
+use zerocaf::traits::ops::Double;
+
+#[derive(Clone)]
+// Represents a Sonny Edwards Point using Twisted Edwards Extended Coordinates
+pub struct SonnyRistrettoPointGadget {
+    pub X: LinearCombination,
+    pub Y: LinearCombination,
+    pub Z: LinearCombination,
+    pub T: LinearCombination,
+}
+
+impl SonnyRistrettoPointGadget {
+    /// Builds a `SonnyRistrettoPointGadget` from a `SonnyRistrettoPoint` adding a constrain
+    /// that checks that the point relies on the curve and another one checking that
+    /// it is indeed a RistrettoPoint.
+    pub fn from_point(point: SonnyRistrettoPoint, cs: &mut dyn ConstraintSystem) -> Self {
+        let gadget_p = SonnyRistrettoPointGadget {
+            X: Scalar::from_bytes_mod_order(point.0.X.to_bytes()).into(),
+            Y: Scalar::from_bytes_mod_order(point.0.Y.to_bytes()).into(),
+            Z: Scalar::from_bytes_mod_order(point.0.Z.to_bytes()).into(),
+            T: Scalar::from_bytes_mod_order(point.0.T.to_bytes()).into(),
+        };
+        gadget_p.ristretto_gadget(cs, Some(point));
+        gadget_p
+    }
+
+    pub fn from_LCs(lcs: Vec<LinearCombination>, cs: &mut ConstraintSystem) -> Self {
```

Comment:
Very minor issue, but should we not follow the snake case?

Reply:
Well, it was more clear that we were refering to LinearCombinations, but I can just allow snake case on this file if the warnings are a problem.

If you think it's better to change it, I'll do it